### PR TITLE
Modifying message on Progress page.

### PR DESCRIPTION
**Pull request by diegomillan**
### Description

This PR is a theme customization in order to change the message on progress lms section. It works together with this [PR](https://github.com/Stanford-Online/edx-platform/pull/528). 
### Reviewers

If you've been tagged for review, please check your corresponding box once you've given the :+1:.
- [ ] Code review: @stvstnfrd
- [ ] Code review: @felipemontoya

FYI: Tag anyone who might be interested in this PR here

@juancamilom
### Post-review
- [ ] Rebase and squash commits

### CODE REVIEW

**Comment by felipemontoya** [comment]
:+1: @stvstnfrd ready for your feedback here

**Comment by stvstnfrd** [comment]
@diegomillan @felipemontoya 

Again, overriding an entire file just to change one line of text is
overly burdensome.

One major issue with overriding entire theme files is that it's an
all-or-nothing affair. In this particular case, it means that if edX
were to make _any_ changes to the progress page's template, we would
never see them (as new features are added, bugs are fixed, etc.). This
has been the cause of much pain in the past, as it then requires you to
proactively monitor the platform template and manually port over changes
each time you merge.  Regardless, you always miss something eventually.
Worse, if the `context` values change, you risk cases where the entire
page would crash until manually updated.

In short, while it may make this one change easier for you (that's
still debatable, though), it gives us a lifetime of maintenance that we
aren't willing to take on here.

Instead, the preferred approach would be to override/theme _only_ that
specific section. We do this with Mako blocks, which can then be
overridden on a page-by-page or instance-by-instance basis.

There are a number of examples of this in the platform code (both our
fork, as well as upstream); here's one:
https://github.com/Stanford-Online/edx-platform/blob/master/lms/templates/main.html#L30-L34

File: templates/courseware/progress.html [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+      % endif
+
+      <header class="progress-certificates">
+        <h1 class="progress-certificates-title">${_("Course Progress for Student '{username}' ({email})").format(username=student.username, email=student.email) | h}</h1>
+      </header>
+
+      <section class="progress-info-mouseover">
+        <p>
+            ${_("""Only pre-test and post-test scores are tracked. You must
+                receive a score of 75% or higher on the post-test in order to eligibilie a certificate.
```
Comment (stvstnfrd) [inline templates/courseware/progress.html]: 75% should not be hard-coded.

Comment (stvstnfrd) [inline templates/courseware/progress.html]: `eligibilie` is misspelled and the sentence is grammatically incorrect.

Did you mean to say "in order to be eligible for a certificate" or maybe
"in order to earn a certificate"?
Please check with your copy text from Mark.

**Comment by felipemontoya** [comment]
Hello @stvstnfrd , I understand your concern of this being more difficult to maintain.

Since this is the same issue for some of the other PRs, I'll comment here, and we can update all the other with the same solution that we come up with at this PR.

About your proposal of using blocks, here it is a bit problematic, since we would have to inherit from a template with the same name as the one we are in.

For example we have a file `courseware/progress.html` in the platform repo

```
<%inherit file="main.html" />

... content of progress.html ....

<%block name="block_we_want_to_override">
... content to override ...
<%/block%>
```

and on the theme, we would add a file `courseware/progress.html`, so that the django view renders our file instead of the regular.

```
<%inherit file="courseware/progress.html" /> ## This way 

<%block name="block_we_want_to_override">
... overriden content ...
<%/block%>
```

Only this, creates a loop, since we mean to override the core templates, but the template loader only returns the theme template.

I'm all ears if you know a better solution for this, but I went throug the mako docs and code and could not find anything to help us here.
The solution I currently see for this would be to use an included template just for that section, that we can override in the theme.

Example:

`courseware/progress.html` in the platform repo

```
<%inherit file="main.html" />

... content of progress.html ....

<%include file="fragment-we-can-overide.html" />  ## instead of the block
```

`courseware/fragment-we-can-overide.html` in the platform repo

```
... some content ...
```

`courseware/fragment-we-can-overide.html` at the theme

```
... CME's content ...
```

Any thoughts on this?

File: templates/courseware/progress.html (hunk 1 above)
Comment (felipemontoya) [reply templates/courseware/progress.html]: Sure thing, we just copied verbatim the copy we got from mark, but we will take a look

**Comment by stvstnfrd** [comment]
@felipemontoya Yeah, we're (unfortunately) familiar with the infinite
loop include issue; in fact, it's the primary issue blocking us from
converting over to CompTheming. We're working on a fix for upstream, but
nothing that we expect to land soon.

Your proposal matches the best idea we've come up with;
just include a themable partial into the page on platform.

So we'll end up w/

in platform:templates/progress.html

```
<%include file="progress-info.html" %>
```

in platform:templates/progress-info.html

```
<section class="progress-info-mouseover">
  <p>
      ${_("""Note: Submitted assignments will appear as bars in the Progress chart.
          Mouse over a bar to see more information about the relevant assignment.
      """)}
  </p>
  </section>
```

in cme-theme:templates/progress-info.html

```
<section class="progress-info-mouseover">
  <p>
    ${_("""Only pre-test and post-test scores are tracked. You must
           receive a score of 75% or higher on the post-test in order to eligibilie a certificate.
    """)}
  </p>
</section>
```

This then would allow one to hide the `section` entirely by defining an
empty block.

**Comment by diegomillan** [comment]
Hi @stvstnfrd, I just finished updating this PR. There is also a new [PR](https://github.com/Stanford-Online/edx-platform/pull/528) on stanford platform repo that works together with this changes. 

I corrected grammar and misspelling problems, but "75%" is still hard coded since that was the only input we got from Mark.

**Comment by felipemontoya** [comment]
@stvstnfrd Since the [PR](https://github.com/Stanford-Online/edx-platform/pull/528) this set of changes required has been merged, we can proceed to finish the review and merge process on this PR.

I'l summarize the current status here. First,as I said the 528 PR is merged, the copy was reviewed by Mark. And the commits have been squashed. I can't edit the tick box, since I am not a maintainer, but please @diegomillan do tick my  :+1: box and the post review squash.

@stvstnfrd, diego mentioned the hard coded "75%" string, but since this is a theme template it would make no sense to make it a configuration value, and to make this a course setting would require changes at the edx-platform. I don't think this is worth the effort. If you think it is needed, please let me know.

**Comment by stvstnfrd** [comment]
@felipemontoya Nah, this is good, merging now..
